android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt

```diff
@@ -0,0 +1,225 @@
+package co.nimblehq.ic.kmm.suv.android.ui.screens.surveyquestions
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import co.nimblehq.ic.kmm.suv.android.R
+import co.nimblehq.ic.kmm.suv.android.extension.placeholder
+import co.nimblehq.ic.kmm.suv.android.ui.theme.AppTheme
+import co.nimblehq.ic.kmm.suv.android.ui.theme.Typography
+import co.nimblehq.ic.kmm.suv.android.util.LoadingParameterProvider
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.HorizontalPager
+
+@Composable
+fun SurveyQuestionsScreen(
+    onCloseClick: () -> Unit
+) {
+    SurveyQuestionsScreenContent(
+        // TODO: Remove dummy data
+        SurveyQuestionsContentUiModel(
+            isLoading = false,
+            backgroundUrl = "https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_l",
+            questions = listOf(
+                QuestionContentUiModel(
+                    "1/2",
+                    "What your name?"
+                ),
+                QuestionContentUiModel(
+                    "2/2",
+                    "How old are you?"
+                )
+            )
+        ),
+        onCloseClick
+    )
+}
+
+@Composable
+private fun SurveyQuestionsScreenContent(
+    uiModel: SurveyQuestionsContentUiModel,
+    onCloseClick: () -> Unit = {}
+) {
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black)
+    ) {
+        Box(modifier = Modifier.fillMaxSize()) {
+            Image(
+                painter = painterResource(id = R.drawable.bg_splash),
+                contentDescription = null,
+                contentScale = ContentScale.FillWidth
+            )
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(
+                        brush = Brush.verticalGradient(
+                            colors = listOf(
+                                Color.Transparent,
+                                Color.Black.copy(alpha = 0.6f)
+                            )
+                        )
+                    )
+            )
+        }
+    }
```

Comment:
I think the whole Image & Box widgets group could be extracted into a shared widget for 3 screens: Home Survey, Survey Detail & Survey Question screens. Could we do a code optimization here?

<img width="642" alt="image" src="https://user-images.githubusercontent.com/16315358/207567160-13d5a566-6783-4f0c-b370-77385c901656.png">

Reply:
Nice catch! Fixed in https://github.com/suho/kmm-ic/pull/114/commits/0d0b04b5795c2cf45387030653649ef22f39b1ef